src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -716,6 +729,36 @@ public static ReadOnlyMemory<char> AsMemory(this string? text, Range range)
                     return IndexOfAnyExcept(span, values[0], values[1], values[2], values[3]);
 
                 default:
+                    if (RuntimeHelpers.IsBitwiseEquatable<T>())
+                    {
+                        if (Unsafe.SizeOf<T>() == sizeof(byte) && values.Length == 5)
+                        {
+                            ref byte valuesRef = ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(values));
+
+                            return SpanHelpers.IndexOfAnyExceptValueType(
+                                ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(span)),
+                                valuesRef,
+                                Unsafe.Add(ref valuesRef, 1),
+                                Unsafe.Add(ref valuesRef, 2),
+                                Unsafe.Add(ref valuesRef, 3),
+                                Unsafe.Add(ref valuesRef, 4),
+                                span.Length);
+                        }
+                        else if (Unsafe.SizeOf<T>() == sizeof(short) && values.Length == 5)
+                        {
+                            ref short valuesRef = ref Unsafe.As<T, short>(ref MemoryMarshal.GetReference(values));
+
+                            return SpanHelpers.IndexOfAnyExceptValueType(
```

Comment:
Are our char-specific/byte-specific implementations that much better than the generalized T-based implementation?  I wonder if we can collapse all of these down.

Reply:
You mean if we should extend the 1-5 value special cases to all `T`s instead of just byte and char?

For these specific ones, I wasn't sure if I should use it for other types here to avoid increasing the size even more.

Reply:
Nevermind, I thought you'd special-cased byte/char to have their own call sites was because you were delegating to two different overloads each strongly typed to either byte or char.  But both of these are already calling to a shared T-based method.